pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java

```diff
@@ -84,6 +87,11 @@ public PartitionAssignment rebalancePartitionAssignment(IdealState idealState, T
     PartitionAssignment newPartitionAssignment = new PartitionAssignment(tableConfig.getTableName());
 
     if (tableConfig.getTableType().equals(CommonConstants.Helix.TableType.REALTIME)) {
+      StreamMetadata streamMetadata = new StreamMetadata(tableConfig.getIndexingConfig().getStreamConfigs());
+      if (streamMetadata.hasHighLevelKafkaConsumerType()) {
+        return newPartitionAssignment;
```

Comment:
Can you add the logging information that the table is using `high level kafka consumer` so that the partition assignment would be empty?